x/oracle/types/msgs.go

```diff
@@ -168,3 +172,115 @@ func (msg MsgDelegateFeedConsent) ValidateBasic() error {
 
 	return nil
 }
+
+// NewMsgUpdateParams will creates a new MsgUpdateParams instance
+func NewMsgUpdateParams(authority, title, description string, keys []string, changes Params) *MsgGovUpdateParams {
+	return &MsgGovUpdateParams{
+		Title:       title,
+		Description: description,
+		Authority:   authority,
+		Keys:        keys,
+		Changes:     changes,
+	}
+}
+
+// Type implements Msg interface
+func (msg MsgGovUpdateParams) Type() string { return sdk.MsgTypeURL(&msg) }
+
+// String implements the Stringer interface.
+func (msg MsgGovUpdateParams) String() string {
+	out, _ := yaml.Marshal(msg)
+	return string(out)
+}
+
+// GetSignBytes implements Msg
+func (msg MsgGovUpdateParams) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners implements Msg
+func (msg MsgGovUpdateParams) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Authority)
+}
+
+// ValidateBasic implements Msg
+func (msg MsgGovUpdateParams) ValidateBasic() error {
+	if err := checkers.ValidateProposal(msg.Title, msg.Description, msg.Authority); err != nil {
+		return err
+	}
+
+	for _, key := range msg.Keys {
+		switch key {
+		case string(KeyVotePeriod):
+			if msg.Changes.VotePeriod == 0 {
+				return fmt.Errorf("oracle parameter VotePeriod must be > 0")
+			}
+
+		case string(KeyVoteThreshold):
+			if msg.Changes.VoteThreshold.LTE(sdk.NewDecWithPrec(33, 2)) {
+				return fmt.Errorf("oracle parameter VoteThreshold must be greater than 33 percent")
+			}
+
+		case string(KeyRewardBands):
+			for _, v := range msg.Changes.RewardBands {
+				if v.RewardBand.GT(sdk.OneDec()) || v.RewardBand.IsNegative() {
+					return fmt.Errorf("oracle parameter RewardBand must be between [0, 1]")
```

Comment:
What's the effect of keeping the reward band between 0 and 1? Does this mean we can only allow a difference of $0.50 around the median?

Reply:
Yeah, its been enforced to be between 0 and 1 all this time so if you think it should be larger I can change it here and in https://github.com/ojo-network/ojo/blob/7a4a4c4208329ee3d204a2f4a162506c1c1f1fa4/x/oracle/types/params.go#L282

Reply:
Let's think about it and open up in a separate PR if we need to. I think having a 0.50 difference is okay since it's per-asset